### PR TITLE
Develop output cuda

### PR DESCRIPTION
**Pull request by mattldawson**
use pregenerated CUDA code in comparisons

### CODE REVIEW

File: test/derivative/CMakeLists.txt [hunk 1]
```diff
@@ -17,7 +17,10 @@ if(ENABLE_LLVM)
 endif()
 
 if(ENABLE_GPU)
-  set(CUDA_SRC CudaJitDeriv.cpp CudaGeneralDeriv.cpp)
+  set(CUDA_SRC CudaJitDeriv.cu CudaGeneralDeriv.cu)
+  #if(ENABLE_COMPILED)
+    #    set(COMPILE_SRC jit.cu jit_flipped.cu general.cu general_flipped.cu)
+    #endif()
```
Comment (sjsprecious) [inline test/derivative/CMakeLists.txt]: probably just remove these lines?

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 3.21)
-project(JIT_TEST LANGUAGES CXX C Fortran)
-set (CMAKE_CXX_STANDARD 17)
+project(JIT_TEST LANGUAGES CXX C Fortran CUDA)
+set (CMAKE_CXX_STANDARD 11)
```
Comment (sjsprecious) [inline CMakeLists.txt:3]: So C++17 does not work for CUDA?

File: test/derivative/CudaGeneralDeriv.cu [hunk 3]
```diff
@@ -1,13 +1,20 @@
 #include <string>
 #include <chrono>
+#include <iostream>
+#include <fstream>
 #include "CudaGeneralDeriv.h"
 #include "ClassicDeriv.h"
+#ifdef USE_COMPILED
+#include "general.cu"
+#include "general_flipped.cu"
```
Comment (sjsprecious) [inline test/derivative/CudaGeneralDeriv.cu:9]: I wonder whether we should include the CUDA files here or link the object files later?

File: CMakeLists.txt (hunk 2 above)
Comment (mattldawson) [reply CMakeLists.txt:3]: yeah, seems not to - both c++17 and c++14

File: test/derivative/CMakeLists.txt (hunk 1 above)
Comment (mattldawson) [reply test/derivative/CMakeLists.txt]: done!

**Comment by mattldawson** [comment]
merge with #15

**Pull request closed without merging**